src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,394 @@
+    );
+
+    console.log('failedEntries', failedEntries);
+
+    switch (failedEntries.length) {
+      case 0:
+        this.snapshotAdvice = <>You snapshotted every possible buff! Perfect!</>;
+        break;
+      case 1:
+        console.log(failedEntries[0].relatedBuffs[0]);
+        console.log(enchanting.HURRICANE_BUFF);
+        if (failedEntries[0].relatedBuffs[0] === enchanting.HURRICANE_BUFF) {
+          this.snapshotAdvice = (
+            <>
+              You snapshotted every possible buff, unless{' '}
+              <SpellLink spell={enchanting.HURRICANE_BUFF} />. <br />
+              You can proc both <SpellLink spell={enchanting.POWER_TORRENT_BUFF} /> and{' '}
+              <SpellLink spell={enchanting.HURRICANE_BUFF} /> by casting{' '}
+              <SpellLink spell={SPELLS.SOUL_HARVEST} /> prepull, with a weapon enchanted with
+              Hurricane, then swap to your normal weapon before entering combat. It has only 15%
+              chance to happen, but it adds ~3.5% haste to your doomguard if you succeed!
+            </>
+          );
+        } else {
+          this.snapshotAdvice = (
+            <>
+              You failed to snapshot a buff. Try to summon your doomguard when all buffs are active.
+            </>
+          );
+        }
+        break;
+      default:
+        this.snapshotAdvice = (
+          <>
+            You failed to snapshot several buffs. Try to summon your doomguard when all buffs are
+            active.
+          </>
+        );
+        break;
+    }
+  }
+  private onDoomguardDamage(event: DamageEvent) {
+    if (this.doomguardSummonData.doomguard.totalDamage === undefined) {
+      this.doomguardSummonData.doomguard.totalDamage = 0;
+    }
+    this.doomguardSummonData.doomguard.totalDamage += event.amount;
+  }
+  private onFightEnd(event: FightEndEvent) {
+    if (
+      this.doomguardSummonData.doomguard.summonTimestamp !== 0 &&
+      this.doomguardSummonData.doomguard.deathTimestamp === 0
+    ) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+    }
+  }
+  private onDoomguardDeath(event: DeathEvent) {
+    if (event.targetID === this.doomguardId) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+      const doomguardDuration =
+        event.timestamp - this.doomguardSummonData.doomguard.summonTimestamp;
+      if (doomguardDuration >= 64000) {
+        this.doomguardSummonData.summonedDemonSummary = (
+          <li>
+            <PerformanceMark perf={QualitativePerformance.Perfect} /> You casted{' '}
+            <SpellLink spell={SPELLS.SUMMON_DOOMGUARD} /> at{' '}
+            {formatDuration(
+              this.doomguardSummonData.doomguard.summonTimestamp - this.owner.fight.start_time,
+            )}
+            , for a full <strong>{formatDuration(doomguardDuration)}</strong> duration and{' '}
+            <strong>{this.doomguardSummonData.doomguard.totalDamage}</strong> damage <br />
```

Comment:
use `formatNumber` here:

```suggestion
            <strong>{formatNumber(this.doomguardSummonData.doomguard.totalDamage)}</strong> damage
```